### PR TITLE
Add Radial Distribution function for DNS

### LINKED ISSUE #587: Add g12 Radial Distribution Function
#### $g_{12}$: Radial Distribution Function

The radial distribution function is given by:

$$
g_{12} = \left(\frac{\eta^2 + r_c^2}{R^2 + r_c^2}\right)^{C_1/2}
$$

Where $C_1$ and $r_c$ are derived based on droplet and turbulence properties.

##### $C_1$: Calculation

$$
C_1 = \frac{y(St)}{\left(|\mathbf{g}| / (v_k / \tau_k)\right)^{f_3(R_\lambda)}}
$$

$$
y(St) = -0.1988 St^4 + 1.5275 St^3 - 4.2942 St^2 + 5.3406 St
$$

$$
f_3(R_\lambda) = 0.1886 \exp\left(\frac{20.306}{R_\lambda}\right)
$$

Where:

$$ St = max(St_1, St_2) $$

Since the fitting for $y(St)$ was done for a limited range of St in DNS,
it should be set to zero for large $St$ when the function $y(St)$ becomes negative.

##### $r_c$: Expression

$$
\left(\frac{r_c}{\eta}\right)^2 = |St_2 - St_1| F(a_o, R_\lambda)
$$

solving for $r_c$:

$$
r_c = \eta \sqrt{|St_2 - St_1| F(a_o, R_\lambda)}
$$

where:

$$
a_{Og} = a_o + \frac{\pi}{8} \left(\frac{|\mathbf{g}|}{v_k / \tau_k}\right)^2
$$

$$
F(a_{Og}, R_\lambda) = 20.115 \left(\frac{a_{Og}}{R_\lambda}\right)^{1/2}
$$

### PR DESCRIPTION
**Pull request by Gorkowski**
Added the radial distribution function
Added some self broadcasting helper functions
 
Fixes #587

## Summary by Sourcery

Implement the radial distribution function g12 for turbulent coagulation using the Ayala et al. (2008) model.  Add helper functions for self-broadcasting operations.

New Features:
- Calculate the radial distribution function g12 for turbulent coagulation.

Tests:
- Add tests for the radial distribution function calculation, including validation of inputs and edge cases.

## Summary by Sourcery

New Features:
- Calculate the radial distribution function g12 for turbulent coagulation.

### CODE REVIEW

**Comment by Gorkowski** [comment]
@sourcery-ai review

File: particula/dynamics/coagulation/turbulent_dns_kernel/tests/g12_radial_distribution_ao2008_test.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    """
+    particle_radius = np.array(
+        [10e-6, 20e-6, 30e-6]
+    )  # Particle radii in meters
+    stokes_number = np.array([0.5, 0.7, 1.2])  # Stokes numbers
+    kolmogorov_length_scale = 1e-6  # m
+    reynolds_lambda = 100  # Taylor-microscale Reynolds number
+    normalized_accel_variance = 11  # Normalized acceleration variance
+    kolmogorov_velocity = 0.01  # m/s
+    kolmogorov_time = 0.005  # s
```
[the remark this answers is not in the document]
Comment (Gorkowski) [reply particula/dynamics/coagulation/turbulent_dns_kernel/tests/g12_radial_distribution_ao2008_test.py:30]: not physical meaningful

File: particula/dynamics/coagulation/turbulent_dns_kernel/g12_radial_distribution_ao2008.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    y_stokes = _compute_y_stokes(stokes_number)
+    f3_lambda = _compute_f3_lambda(reynolds_lambda)
+    gravity_term = np.abs(STANDARD_GRAVITY) / (
+        kolmogorov_velocity / kolmogorov_time
+    )
+
+    return y_stokes / (gravity_term**f3_lambda)
+
+
+def _compute_y_stokes(
```
[the remark this answers is not in the document]
Comment (Gorkowski) [reply particula/dynamics/coagulation/turbulent_dns_kernel/g12_radial_distribution_ao2008.py:130]: we'll see if it is a problem first
